pkg/cloudclient/gcp/private.go

```diff
@@ -115,10 +115,28 @@ func (c *Client) setDefaultAPIPublic(ctx context.Context, kclient client.Client,
 	return nil
 }
 
-func (c *Client) ensureDNSForService(kclient client.Client, svc *corev1.Service, dnsName string) error {
+func (c *GCPClient) ensureDNSForService(kclient client.Client, svc *corev1.Service, dnsName string) error {
 	// google.golang.org/api/dns/v1.Service is a struct, not an interface, which
 	// will make this all but impossible to write unit tests for
 
+	// Forwarding rule is necessary for rh-api lb setup
+	// Check forwarding rule exists first
+	if svc.Spec.Type == "LoadBalancer" {
+		ingressList := svc.Status.LoadBalancer.Ingress
+		if len(ingressList) == 0 {
+			// the LB doesn't exist
+			return cioerrors.NewLoadBalancerNotReadyError()
+		}
+		rhapiLbIP := ingressList[0].IP
+		// ensure forwarding rule exists in GCP for service
+		_, err := findGCPForwardingRuleForExtIP(c, rhapiLbIP)
```

Comment:
There should also be a path to handle if the returned value is an empty struct.

Reply:
findGCPForwardingRuleForExtIP will return nil instead of an empty struct since its return value is a pointer. But that still needed handling. Added missing check. Thanks for spotting that.